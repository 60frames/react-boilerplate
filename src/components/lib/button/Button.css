.root {
    border: 1px solid #ccc;
    border-radius: 3px;
    background-color: #fff;
    background: #fff;
    background: linear-gradient(to bottom, rgb(255, 255, 255) 0%,rgb(219, 219, 219) 100%);
    cursor: pointer;
    font-size: 16px;
    padding: 8px 16px;
}
