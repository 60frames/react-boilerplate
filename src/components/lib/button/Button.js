import React, { PropTypes } from 'react';
import styles from 'components/lib/button/Button.css';

function Button({ onClick, children }) {
    return (
        <button className={styles.root} onClick={onClick}>
            {children}
        </button>
    );
}

Button.propTypes = {
    children: PropTypes.string.isRequired,
    onClick: PropTypes.func,
};

export default Button;
