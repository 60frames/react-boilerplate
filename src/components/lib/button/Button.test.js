/* eslint-env jasmine */

import React from 'react';
import renderer from 'react-test-renderer';
import Button from 'components/lib/button/Button';

describe('components/lib/button/Button', () => {

    it('renders correctly', () => {
        const component = renderer.create(
            <Button onClick={() => {}}>
                Hello Button.
            </Button>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
