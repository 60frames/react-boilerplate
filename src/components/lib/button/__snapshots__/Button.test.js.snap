exports[`components/lib/button/Button renders correctly 1`] = `
<button
  className="root"
  onClick={[Function]}>
  Hello Button.
</button>
`;
